assets/javascripts/discourse/services/chat-draft-handler.js

```diff
@@ -0,0 +1,48 @@
+import Service from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import { bind } from "discourse-common/utils/decorators";
+
+const DEFAULT_DRAFT = {
+  value: "",
+  uploads: [],
+  replyToMsg: null,
+};
+
+export default class ChatDraftHandler extends Service {
+  init() {
+    super.init(...arguments);
+
+    this._draftStore = {};
+    if (this.currentUser.chat_drafts) {
+      this.currentUser.chat_drafts.forEach((draft) => {
+        this._draftStore[draft.channel_id] = JSON.parse(draft.data);
+      });
+    }
+  }
+
+  @bind
```

Comment:
what is the bind doing here?

Reply:
If this is for the debounce, I would recommend you have a debounce handler in chat-live-pane which would call this instead of directly debouncing the call to the draft handler

Reply:
No, actually I wasn't able to access `this._draftStore` in these methods. I looked around and figured that `@bind` fixed it.

Reply:
I realized that bind isn't needed and removed it. But I promise things weren't working earlier without it. Not sure what that was.